packages/toolpad-app/src/toolpad/AppEditor/ErrorUpgrade.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import Alert from '@mui/material/Alert';
+import Button from '@mui/material/Button';
+import OpenInNewIcon from '@mui/icons-material/OpenInNew';
+import { UPGRADE_URL } from '../../constants';
+
+export function ErrorUpgrade({ feature, hideAction }: { feature?: string; hideAction?: boolean }) {
+  return (
+    <Alert
+      severity="error"
+      action={
+        hideAction ? null : (
+          <Button variant={'text'} LinkComponent={'a'} href={UPGRADE_URL} target="_blank">
+            Upgrade
+            <OpenInNewIcon fontSize="inherit" sx={{ ml: 1 }} />
+          </Button>
+        )
+      }
+    >
+      {`${feature ?? 'This feature'} is only available on paid plans.`}
```

Comment:
Oh another nit, the error message right now says "Roles is only available on paid plans." instead of "Roles are". Also not sure if worth correcting though as we'd need to use some `pluralize` function or something, just thought I'd mention it for completeness.

Reply:
Agree - I've changed these labels to be singular nouns everywhere for now! We can deal with detecting pluralisation later

Reply:
Agreed! For now this is using singular nouns everywhere but we can enhance this to detect pluralization